internal/pkg/propagation/summaries.go

```diff
@@ -0,0 +1,761 @@
+		ifTainted:   0b100,
+		taintedArgs: []int{1},
+	},
+	// func HTMLEscape(w io.Writer, b []byte) {
+	"html/template.HTMLEscape": {
+		ifTainted:   0b10,
+		taintedArgs: []int{0},
+	},
+	// func HTMLEscapeString(s string) string {
+	"html/template.HTMLEscapeString": fromFirstArgToFirstRet,
+	// func HTMLEscaper(args ...interface{}) string {
+	"html/template.HTMLEscaper": fromFirstArgToFirstRet,
+	// func JSEscape(w io.Writer, b []byte) {
+	"html/template.JSEscape": {
+		ifTainted:   010,
+		taintedArgs: []int{0},
+	},
+	// func JSEscapeString(s string) string {
+	"html/template.JSEscapeString": fromFirstArgToFirstRet,
+	// func JSEscaper(args ...interface{}) string {
+	"html/template.JSEscaper": fromFirstArgToFirstRet,
+	// func URLQueryEscaper(args ...interface{}) string {
+	"html/template.URLQueryEscaper": fromFirstArgToFirstRet,
+	// func Clean(path string) string {
+	"path.Clean": fromFirstArgToFirstRet,
+	// func Split(path string) (dir, file string) {
+	"path.Split": {
+		ifTainted:   0b1,
+		taintedRets: []int{0, 1},
+	},
+	// func Join(elem ...string) string {
+	"path.Join": fromFirstArgToFirstRet,
+	// func Base(path string) string {
+	"path.Base": fromFirstArgToFirstRet,
+	// func Clean(path string) string {
+	"path/filepath.Clean": fromFirstArgToFirstRet,
+	// func ToSlash(path string) string {
+	"path/filepath.ToSlash": fromFirstArgToFirstRet,
+	// func FromSlash(path string) string {
+	"path/filepath.FromSlash": fromFirstArgToFirstRet,
+	// func SplitList(path string) []string {
+	"path/filepath.SplitList": fromFirstArgToFirstRet,
+	// func Split(path string) (dir, file string) {
+	"path/filepath.Split": {
+		ifTainted:   0b1,
+		taintedRets: []int{0, 1},
+	},
+	// func Join(elem ...string) string {
+	"path/filepath.Join": fromFirstArgToFirstRet,
+	// func Ext(path string) string {
+	"path/filepath.Ext": fromFirstArgToFirstRet,
+	// func Abs(path string) (string, error) {
+	"path/filepath.Abs": fromFirstArgToFirstRet,
+	// func Base(path string) string {
+	"path/filepath.Base": fromFirstArgToFirstRet,
+	// func New(out io.Writer, prefix string, flag int) *Logger {
+	"log.New": fromFirstArgToFirstRet,
+	// func (l *Logger) SetOutput(w io.Writer) {
+	"(*log.Logger).SetOutput": {
+		ifTainted:   0b10,
+		taintedArgs: []int{0},
+	},
+	// func (l *Logger) Writer() io.Writer {
+	"(*log.Logger).Writer": fromFirstArgToFirstRet,
+}
+
+// funcKey represents an interface function by its name and its signature.
+// The signature is a string representation containing only the types of
+// the arguments and return values.
+type funcKey struct {
+	name, signature string
+}
+
+// interfaceFuncSummaries contains summaries for common interface functions
+// such as Write or Read, that could be called statically (i.e. a call to
+// a concrete method whose signature matches an interface method) or dynamically
+// (i.e. a call to an interface method on an interface value).
+var interfaceFuncSummaries = map[funcKey]summary{
```

Comment:
My understanding of the current code is that `arg0` is the receiver for all `ifTainted` uses below.  That would be a good comment here.

Reply:
Done.